File: suite/FactorialAccumulator/FactorialAccumulator.rs
fn factorial(a: i64, i: i64) -> i64 {
    if i == 0 {
        a
    } else {
        factorial((i * a) % 1000000007, i - 1)
    }
}

fn main_loop(iters: u64, n: i64) {
    let mut res = factorial(1, n);
    for _ in 1..iters {
        res = factorial(1, n);
    }
    println!("{}", res);
}

fn main() {
    let mut args = std::env::args();
    args.next();
    let iters = args
        .next()
        .expect("Missing Argument iterations")
        .parse::<u64>()
        .expect("Iterations must be a number");
    let n = args
        .next()
        .expect("Missing Argument n")
        .parse::<i64>()
        .expect("n must be a number");
    main_loop(iters, n)
}

File: suite/FactorialAccumulator/FactorialAccumulator.ml
let rec factorial a i  =
  if i = 0 then a
  else factorial ((i * a) mod 1000000007) (i - 1)

let rec main_loop iters n =
  let res = factorial 1 n in
  if iters = 1 then
    print_endline (string_of_int res)
  else main_loop (iters - 1) n

let main =
  let iters = int_of_string Sys.argv.(1) in
  let n = int_of_string Sys.argv.(2) in
  main_loop iters n
